File: problems/104. Maximum Depth of Binary Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.getHeight(0, root)

    def getHeight(self, height: int, root: Optional[TreeNode]) -> int:
        if not root: return 0
        left = self.getHeight(height, root.left) if root.left else 0
        right = self.getHeight(height, root.right) if root.right else 0
        return max(left, right) + 1

File: problems/104. Maximum Depth of Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        return getHeight(0, root);
    }

    private int getHeight(int height, TreeNode root) {
        if (root == null) return 0;
        int left = root.left == null ? 0 : getHeight(height, root.left);
        int right = root.right == null ? 0 : getHeight(height, root.right);
        return Math.max(left, right) + 1;
    }
}